aws-rds-dbcluster/src/main/java/software/amazon/rds/dbcluster/BaseHandlerStd.java

```diff
@@ -458,8 +471,17 @@ protected ProgressEvent<ResourceModel, CallbackContext> setDefaultVpcSecurityGro
         }
     }
 
-    protected boolean shouldSetDefaultVpcSecurityGroupIds(final ProgressEvent<ResourceModel, CallbackContext> progress) {
-        final ResourceModel resourceModel = progress.getResourceModel();
-        return CollectionUtils.isEmpty(resourceModel.getVpcSecurityGroupIds());
+    protected boolean shouldSetDefaultVpcSecurityGroupIds(final ResourceModel previousState, final ResourceModel desiredState) {
+        if (previousState != null) {
+            final List<String> previousVpcIds = CollectionUtils.isEmpty(previousState.getVpcSecurityGroupIds()) ?
+                    ImmutableList.of() : previousState.getVpcSecurityGroupIds();
```

Comment:
NIT: This is a variadic function returning a "real" list instance. Please prefer `Collections.emptyList()`.